src/components/badges/List.tsx

```diff
@@ -67,33 +39,16 @@ function BadgeListSuspended() {
       <ConfettiIfNeeded />
       <Scrollbar>
         <div className={space('space-y-2')}>
-          <BadgeSection
-            title="Mainnet NFT derivatives"
-            minted={ownedExternalERC721DerivativeContracts}
-            proofs={proofsAvailableToMint.filter(
-              (proof) =>
-                proof instanceof ERC721Proof &&
-                proof.network === Network.Mainnet
-            )}
-            onMinted={onMinted}
-          />
-          <BadgeSection
-            title="Goerli NFT derivatives"
-            minted={ownedERC721DerivativeContracts}
-            proofs={proofsAvailableToMint.filter(
-              (proof) =>
-                proof instanceof ERC721Proof && proof.network === Network.Goerli
-            )}
-            onMinted={onMinted}
-          />
-          <BadgeSection
-            title="Email derivatives"
-            minted={ownedEmailDerivativeContracts}
-            proofs={proofsAvailableToMint.filter(
-              (proof) => proof instanceof EmailProof
-            )}
-            onMinted={onMinted}
-          />
+          {(Object.keys(data) as DataKey[]).map((ledgerName) => (
+            <BadgeSection
+              title={badgeConfig[data[ledgerName].badgeType].title(
```

Comment:
Feels like we need to simplify this construction, maybe we can have a helper to extract the badge type from data?

Reply:
The helper won't reduce the size of this, maybe simplification of the `data` and `badgeConfig` can help us? For example, can we extract `badgeConfig` using `ledgerName`?

Reply:
Okay, we'll have to think about it, let's fix it later